### PR TITLE
[DSI-6467] - Added extensive health checks for AFD

### PR DESCRIPTION
**Pull request by ipredoi**
**Jira ticket**:https://dfe-secureaccess.atlassian.net/browse/DSI-6467

**Changes:**

- Implemented health checks for API services.
- implemented health checks for all identifying parties and oidc 
- Expanded Sequelize connection checks to validate the health of all associated databases.
- Extended Redis checks to validate the health of all connections.
- updated Node Js version to 18.17.0
- fixed vulnerabilities
- fixed linting errors
- added unit tests coverage

**Checks performed summary:** 
https://dfe-secureaccess.atlassian.net/wiki/spaces/NSA/pages/3669426177/Healhtcheck+library

### CODE REVIEW

File: lib/identifyingPartyHealthCheck.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        status: constants.HEALTY_STATUS_MESSAGE,
+      };
+    } catch (error) {
+      return {
+        key,
+        status: error.toString(),
+      };
+    }
+  } else {
+    return null;
```
Comment (malikimran34) [inline lib/identifyingPartyHealthCheck.js:25]: what is the outcome if its null

File: lib/redisCheck.js [hunk 2]
```diff
@@ -1,65 +1,75 @@
...
-          client.disconnect()
-        } catch (e) {
-          // swallow this as we expect disconnects to fail if connect also failed.
-        }
+    const client = new Redis(connectionString);
+    client.on('ready', () => {
+      try {
+        client.disconnect();
+      } catch (e) {
+        // log here if we start seeing leaks
```
Comment (malikimran34) [inline lib/redisCheck.js]: add a log message

**Review by malikimran34 — APPROVED** [review]

File: lib/identifyingPartyHealthCheck.js (hunk 1 above)
Comment (ipredoi) [reply lib/identifyingPartyHealthCheck.js:25]: if return null, that particular dependency won't be checked

File: lib/redisCheck.js (hunk 2 above)
Comment (ipredoi) [reply lib/redisCheck.js]: log added

**Pull request merged**